core/src/test/scala/cromwell/core/retry/RetrySpec.scala

```diff
@@ -0,0 +1,92 @@
+package cromwell.core.retry
+
+import akka.actor.ActorSystem
+import cromwell.core.CromwellFatalException
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.time.{Millis, Seconds, Span}
+import org.scalatest.{FlatSpec, Matchers}
+import Retry._
+import scala.concurrent.duration._
+import scala.concurrent.Future
+
+class RetrySpec extends FlatSpec with Matchers with ScalaFutures {
+  implicit val actorSystem = ActorSystem("retry-spec")
+
+  class TransientException extends Exception
+  class MockWork(n: Int, transients: Int = 0) {
+    implicit val ec = actorSystem.dispatcher
+
+    var counter = n
+
+    def doIt(): Future[Int] = {
+      if (counter == 0)
+        Future.successful(9)
+      else {
+        counter -= 1
+        val ex = if (counter <= transients) new TransientException else new IllegalArgumentException("Failed")
+        Future.failed(ex)
+      }
+    }
+  }
+
+  implicit val defaultPatience = PatienceConfig(timeout = Span(30, Seconds), interval = Span(100, Millis))
+
+  private def runRetry(retries: Int,
+                       work: MockWork,
+                       isTransient: Throwable => Boolean = Retry.throwableToFalse,
```

Comment:
huh?  why not just lambda this here rather than add the method to production code?